src/drafts/utils/character-coordinates.ts

```diff
@@ -1,66 +1,198 @@
-import getCaretCoordinates from '@koddsson/textarea-caret'
-
-export type Coordinates = {
+export type CharacterCoordinates = {
+  /** Number of pixels from the origin down to the top edge of the character. */
   top: number
+  /** Number of pixels from the origin right to the left edge of the character. */
   left: number
+  /** Height of the character. */
+  height: number
 }
 
+// Note that some browsers, such as Firefox, do not concatenate properties
+// into their shorthand (e.g. padding-top, padding-bottom etc. -> padding),
+// so we have to list every single property explicitly.
+const propertiesToCopy = [
+  'direction', // RTL support
+  'boxSizing',
+  'width', // on Chrome and IE, exclude the scrollbar, so the mirror div wraps exactly as the textarea does
+  'height',
+  'overflowX',
+  'overflowY', // copy the scrollbar for IE
+
+  'borderTopWidth',
+  'borderRightWidth',
+  'borderBottomWidth',
+  'borderLeftWidth',
+  'borderStyle',
+
+  'paddingTop',
+  'paddingRight',
+  'paddingBottom',
+  'paddingLeft',
+
+  // https://developer.mozilla.org/en-US/docs/Web/CSS/font
+  'fontStyle',
+  'fontVariant',
+  'fontWeight',
+  'fontStretch',
+  'fontSize',
+  'fontSizeAdjust',
+  'lineHeight',
+  'fontFamily',
+
+  'textAlign',
+  'textTransform',
+  'textIndent',
+  'textDecoration', // might not make a difference, but better be safe
+
+  'letterSpacing',
+  'wordSpacing',
+
+  'tabSize',
+  'MozTabSize' as 'tabSize' // prefixed version for Firefox <= 52
+] as const
+
 /**
  * Obtain the coordinates (px) of the bottom left of a character in an input, relative to the
- * top-left corner of the input itself.
- * @param input The target input element.
- * @param index The index of the character to calculate for.
- * @param adjustForScroll Control whether the returned value is adjusted based on scroll position.
+ * top-left corner of the interior of the input (not adjusted for scroll).
+ *
+ * Adapted from https://github.com/koddsson/textarea-caret-position, which was itself
+ * forked from https://github.com/component/textarea-caret-position.
```

Comment:
I'm no expert on licensing. Both source projects are MIT-licensed - is this sufficient attribution? Do we also need to pull in the copyright statements from those licenses?

Reply:
I _think_ this is fine 👍